odoo/addons/base/models/ir_actions_report.py

```diff
@@ -489,6 +489,9 @@ def barcode(self, barcode_type, value, width=600, height=100, humanreadable=0):
             barcode_type = 'EAN13'
             if len(value) in (11, 12):
                 value = '0%s' % value
+        elif barcode_type == 'auto':
+            symbology_guess = {8: 'EAN8', 13: 'EAN13'}
+            barcode_type = symbology_guess.get(len(value), 'Code128')
```

Comment:
That seems like an overly complicated way to implement a conditional.

Reply:
Well,I find it more readable than
`barcode_type = len(value) == 8 and 'EAN8' or len(value) == 13 and 'EAN13' or 'Code128' `
or 
`barcode_type = 'EAN8' if len(value) == 8 else 'EAN13' if len(value) == 13 else 'Code128'`

Also, I didn't want to add more nested level of `if/elif/else`.

Finally, I think it's easier to extend it if an barcode type I don't know is needed.

But I can change it if that burns eyes.

Reply:
fair 'nuf